t/rest_server/.github/workflows/deploy.yaml

```diff
@@ -0,0 +1,106 @@
+# Copyright 2023 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+name: 'deploy'
+
+on:
+  workflow_dispatch:
+  workflow_call:
+
+concurrency:
+  group: '${{ github.workflow }}-${{ github.head_ref || github.ref }}'
+  cancel-in-progress: true
+
+env:
+  # This should be the only env variable defined in this file. Define all other
+  # variables in the bootstrap file.
+  # More environments are expected to be added in the future.
+  BOOTSTRAP_ENV_FILE: '.github/config/autopush.env'
+
+jobs:
+  image-release:
+    permissions:
+      contents: 'read'
+      id-token: 'write'
+    name: 'Release Image'
+    runs-on: 'ubuntu-latest'
+    steps:
+      - name: 'Checkout Repo'
+        uses: 'actions/checkout@8e5e7e5ab8b370d6c329ec480221332ada57f0ab' # ratchet:actions/checkout@v3
+      - name: 'Load variables from bootstrap .env file'
+        run: |
+          # Filter out comment lines, empty lines, and lines with only whitespace.
+          grep -v '^[[:space:]]*#\|^$\|^\s*$' ${{ env.BOOTSTRAP_ENV_FILE }} >> ${GITHUB_ENV}
+      - name: 'Setup Go'
+        uses: 'actions/setup-go@4d34df0c2316fe8122ab82dc22947d607c0c91f9' # ratchet:actions/setup-go@v4
+        with:
+          go-version: '1.20.2'
+      - name: 'Build Go binary'
+        run: |
+          CGO_ENABLED=0 GOOS=linux go build -o restserver .
+      - name: 'Build Docker Image'
```

Comment:
I would move the build go binary and build docker image below the auth step, it would be more efficient to fail fast with auth rather than spending time building the container only to fail in the auth step.

Reply:
Moved below auth.